examples/99_in_tool_selection/ex_in_tool_selection/main.rb

```diff
@@ -0,0 +1,364 @@
+# Copyright 2021 Trimble Inc
+# Licensed under the MIT license
+
+require 'ex_in_tool_selection/inference_lock_helper'
+
+module Examples
+  module InToolSelection
+
+    class ComponentMoveTool
+
+      # Threshold in logical screen pixels for when the mouse is considered to
+      # be dragged.
+      DRAG_THRESHOLD = 10
+
+      def activate
+        @mouse_ip = Sketchup::InputPoint.new
+        @picked_first_ip = Sketchup::InputPoint.new
+        @mouse_position = ORIGIN
+        @mouse_down = ORIGIN
+        @distance = 0
+
+        # Create a InferenceLockHelper object to use in this tool.
+        # See inference_lock_helper.rb for details.
+        @inference_lock_helper = InferenceLockHelper.new
+
+        # Empty the selection if the tool can't use it.
+        # Otherwise keep this preselection.
+        Sketchup.active_model.selection.clear unless valid_selection?
+
+        # The tool selection is tracked separately from the SketchUp selection.
+        # When we are in the selection stage we want to temporarily select
+        # hovered entities for previewing them as selected, but our tool logic
+        # treats the selection as empty until the user clicks.
+        #
+        # Native tools can preview entities as selected without actually
+        # adding them to the selection, but the Ruby API doesn't support this.
+        @selection = Sketchup.active_model.selection.to_a
+
+        # Remember whether tool was activated with preselection or not as this
+        # affect how the tool resets.
+        @preselection = !Sketchup.active_model.selection.empty?
```

Comment:
A more accurate name could be `@preselected`?

Reply:
Switching to `@preselecetd`